monai/transforms/spatial/array.py

```diff
@@ -293,24 +293,18 @@ def __call__(  # type: ignore
         dtype = dtype or self.dtype
         src_affine = src_meta.get("affine")
         dst_affine = dst_meta.get("affine")
-        ndim = len(img.shape[1:])
-        spatial_size = dst_meta.get("dim", [])[1 : ndim + 2]
         img, updated_affine = super().__call__(
             img=img,
             src_affine=src_affine,
             dst_affine=dst_affine,
-            spatial_size=spatial_size,
+            spatial_size=dst_meta.get("spatial_shape"),
             mode=mode,
             padding_mode=padding_mode,
             align_corners=align_corners,
             dtype=dtype,
         )
         dst_meta = deepcopy(dst_meta)
         dst_meta["affine"] = updated_affine
-        if "dim" in dst_meta:
-            dst_meta["dim"] = src_meta.get("dim", [])
-        if "pixdim" in dst_meta:
-            dst_meta["pixdim"] = src_meta.get("pixdim", [])
```

Comment:
If these aren't updated, then they won't be consistent with the new image surely?

Reply:
The unit test first saves the image and then loads it back in before checking the meta data is correct. This works because the savers will update the metadata, but really this should be updated by the transform.

Reply:
that's a good point, the saver might have restored the resampling. I updated the tests to have the comparison with the resampled meta... https://github.com/Project-MONAI/MONAI/pull/3847/commits/12bb4a28672be301cce4d8017e6b5c13c278325a

so in my understanding
```py
ResampleToMatch()(data["im2"], 
                  src_meta=data["im2_meta_dict"], 
                  dst_meta=data["im1_meta_dict"])
```
the im2 will be processed and saved in `im1_meta_dict` for after this transform, perhaps you are thinking about resuming the im2_meta_dict at some point (when saving it)?

Reply:
The way I envisaged this transform is that `im2` would be resampled such that it is aligned with `im1_meta_dict`. The problem is that although we'll update the image, we don't update the meta data, which is what I was attempting to do with the 4 lines that have been removed. I don't think we should be doing anything like `im2_meta_dict = deepcopy(im1_meta_dict)` because it's only the spatial aspects that have been updated.

Reply:
Just added some tests to your PR and the meta data seems to be being updated so all good! Sorry about that.

Reply:
thanks, no worries, actually it's a good point for discussion,
monai reader/writer specifically handles `original_affine` and that's the 'raw' affine
https://github.com/Project-MONAI/MONAI/blob/d6327a765cf23dafd680b2a4b472edf9639e1087/monai/data/image_reader.py#L264

https://github.com/Project-MONAI/MONAI/blob/d6327a765cf23dafd680b2a4b472edf9639e1087/monai/data/image_writer.py#L324

now after `ResampleToMatch` not sure what to put at `original_affine`. this request seems to want to avoid the original affines ones https://github.com/Project-MONAI/MONAI/discussions/3842